### PR TITLE
Deprecate linear algebra wrappers

### LINKED ISSUE #46: Deprecate linear algebra functions
Maintaining wrappers for basic linear algebra functions does not make sense. Use the approach from [SPICE.jl](https://github.com/JuliaAstro/SPICE.jl/blob/master/src/m.jl#L279): Deprecate the functions and point towards sensible pure-Julia alternatives.

* [x] [anp](https://github.com/liberfa/erfa/blob/master/src/anp.c) - Normalize angle into the range 0 <= a < 2pi.
* [X] ~[anpm](https://github.com/liberfa/erfa/blob/master/src/anpm.c) - Normalize angle into the range -pi <= a < +pi.~
* [x] [cp](https://github.com/liberfa/erfa/blob/master/src/cp.c) - Copy a p-vector.
* [x] [cpv](https://github.com/liberfa/erfa/blob/master/src/cpv.c) - Copy a position/velocity vector.
* [x] [cr](https://github.com/liberfa/erfa/blob/master/src/cr.c) - Copy an r-matrix.
* [x] [ir](https://github.com/liberfa/erfa/blob/master/src/ir.c) - Initialize an r-matrix to the identity matrix.
* [x] [p2pv](https://github.com/liberfa/erfa/blob/master/src/p2pv.c) - Extend a p-vector to a pv-vector by appending a zero velocity.
* [x] [pdp](https://github.com/liberfa/erfa/blob/master/src/pdp.c) - p-vector inner (=scalar=dot) product.
* [x] [pm](https://github.com/liberfa/erfa/blob/master/src/pm.c) - Modulus of p-vector.
* [x] [pmp](https://github.com/liberfa/erfa/blob/master/src/pmp.c) - P-vector subtraction.
* [x] [pn](https://github.com/liberfa/erfa/blob/master/src/pn.c) - Convert a p-vector into modulus and unit vector.
* [x] [ppp](https://github.com/liberfa/erfa/blob/master/src/ppp.c) - P-vector addition.
* [x] [ppsp](https://github.com/liberfa/erfa/blob/master/src/ppsp.c) - P-vector plus scaled p-vector.
* [x] [pv2p](https://github.com/liberfa/erfa/blob/master/src/pv2p.c) - Discard velocity component of a pv-vector.
* [x] ~[pvdpv](https://github.com/liberfa/erfa/blob/master/src/pvdpv.c) - Inner (=scalar=dot) product of two pv-vectors.~
* [x] [pvm](https://github.com/liberfa/erfa/blob/master/src/pvm.c) - Modulus of pv-vector.
* [x] [pvmpv](https://github.com/liberfa/erfa/blob/master/src/pvmpv.c) - Subtract one pv-vector from another.
* [x] [pvppv](https://github.com/liberfa/erfa/blob/master/src/pvppv.c) - Add one pv-vector to another.
* [X] ~[pvu](https://github.com/liberfa/erfa/blob/master/src/pvu.c) - Update a pv-vector.~
* [X] ~[pvup](https://github.com/liberfa/erfa/blob/master/src/pvup.c) - Update a pv-vector, discarding the velocity component.~
* [x] ~[pvxpv](https://github.com/liberfa/erfa/blob/master/src/pvxpv.c) - Outer (=vector=cross) product of two pv-vectors.~
* [x] [pxp](https://github.com/liberfa/erfa/blob/master/src/pxp.c) - p-vector outer (=vector=cross) product.
* [x] ~[rx](https://github.com/liberfa/erfa/blob/master/src/rx.c) - Rotate an r-matrix about the x-axis.~
* [x] [rxp](https://github.com/liberfa/erfa/blob/master/src/rxp.c) - Multiply a p-vector by an r-matrix.
* [x] [rxpv](https://github.com/liberfa/erfa/blob/master/src/rxpv.c) - Multiply a pv-vector by an r-matrix.
* [x] [rxr](https://github.com/liberfa/erfa/blob/master/src/rxr.c) - Multiply two r-matrices.
* [x] ~[ry](https://github.com/liberfa/erfa/blob/master/src/ry.c) - Rotate an r-matrix about the y-axis.~
* [x] ~[rz](https://github.com/liberfa/erfa/blob/master/src/rz.c) - Rotate an r-matrix about the z-axis.~
* [x] [s2xpv](https://github.com/liberfa/erfa/blob/master/src/s2xpv.c) - Multiply a pv-vector by two scalars.
* [x] [sxp](https://github.com/liberfa/erfa/blob/master/src/sxp.c) - Multiply a p-vector by a scalar.
* [x] [sxpv](https://github.com/liberfa/erfa/blob/master/src/sxpv.c) - Multiply a pv-vector by a scalar.
* [x] [tr](https://github.com/liberfa/erfa/blob/master/src/tr.c) - Transpose an r-matrix.
* [x] [trxp](https://github.com/liberfa/erfa/blob/master/src/trxp.c) - Multiply a p-vector by the transpose of an r-matrix.
* [x] [trxpv](https://github.com/liberfa/erfa/blob/master/src/trxpv.c) - Multiply a pv-vector by the transpose of an r-matrix.
* [x] [zp](https://github.com/liberfa/erfa/blob/master/src/zp.c) - Zero a p-vector.
* [x] [zpv](https://github.com/liberfa/erfa/blob/master/src/zpv.c) - Zero a pv-vector.
* [x] [zr](https://github.com/liberfa/erfa/blob/master/src/zr.c) - Initialize an r-matrix to the null matrix.

### PR DESCRIPTION
**Pull request by helgee**
Fix #46

### CODE REVIEW

File: src/c.jl [hunk 1]
```diff
@@ -1042,3 +1042,84 @@ function c2ixys(x, y, s)
...
+
+### Given ###
+
+- `p`: p-vector to be copied
+
+### Returned ###
+
+- `c`: copy
+"""
+erfa_cp
```
Comment (helgee) [inline src/c.jl:1061]: I do not want to pirate `Base.cp(::Array{Float64})`. Any better ideas?

Comment (giordano) [reply src/c.jl:1061]: No :disappointed:

**Comment by helgee** [comment]
Will merge by Sunday evening 😉

**Review by giordano — APPROVED** [review]
Ooops, sorry for the late review.  I skimmed through the code, I didn't see any glaring issue :slightly_smiling_face:

**Comment by helgee** [comment]
Great. Thanks 👍

**Pull request merged**